From 21b89d16e2ca79db212587bfea4aa18470984c90 Mon Sep 17 00:00:00 2001
From: Jose Quaresma <quaresma.jose@gmail.com>
Date: Sat, 24 Apr 2021 10:34:47 +0100
Subject: [PATCH] tests: use a dictionaries for environment

meson environment() can't be passed to configure_file and it is needed for installed_tests,
use a dictionary as this is simplest solution to install the environment.

Upstream-Status: Submitted [https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/789]

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>
---
 tests/check/meson.build | 21 +++++++++++++--------
 1 file changed, 13 insertions(+), 8 deletions(-)

diff --git a/tests/check/meson.build b/tests/check/meson.build
index 9a8c42c..6bb346a 100644
--- a/tests/check/meson.build
+++ b/tests/check/meson.build
@@ -161,14 +161,19 @@ foreach t : core_tests
       install: installed_tests_enabled,
     )
 
-    env = environment()
-    env.set('GST_PLUGIN_PATH_1_0', meson.project_build_root())
-    env.set('GST_PLUGIN_SYSTEM_PATH_1_0', '')
-    env.set('GST_STATE_IGNORE_ELEMENTS', '')
-    env.set('CK_DEFAULT_TIMEOUT', '20')
-    env.set('GST_REGISTRY', '@0@/@1@.registry'.format(meson.current_build_dir(), test_name))
-    env.set('GST_PLUGIN_SCANNER_1_0', gst_scanner_dir + '/gst-plugin-scanner')
-    env.set('GST_PLUGIN_LOADING_WHITELIST', 'gstreamer')
+    # meson environment object can't be passed to configure_file and
+    # installed tests uses configure_file to install the environment.
+    # use a dictionary as this is the simplest solution
+    # to install the environment.
+    env = {
+      'GST_PLUGIN_PATH_1_0': meson.project_build_root(),
+      'GST_PLUGIN_SYSTEM_PATH_1_0': '',
+      'GST_STATE_IGNORE_ELEMENTS': '',
+      'CK_DEFAULT_TIMEOUT': '20',
+      'GST_REGISTRY': '@0@/@1@.registry'.format(meson.current_build_dir(), test_name),
+      'GST_PLUGIN_SCANNER_1_0': gst_scanner_dir + '/gst-plugin-scanner',
+      'GST_PLUGIN_LOADING_WHITELIST': 'gstreamer',
+    }
 
     if installed_tests_enabled
       test_conf = configuration_data()
